vizro-core/src/vizro/models/_controls/filter.py

```diff
@@ -335,3 +335,25 @@ def _get_options(
         # changes. See https://pandas.pydata.org/docs/whatsnew/v2.1.0.html#whatsnew-210-enhancements-new-stack.
         targeted_data = pd.concat([targeted_data, pd.Series(current_value)]).stack().dropna()  # noqa: PD013
         return sorted(set(targeted_data) - {ALL_OPTION})
+
+    def _get_proposed_targets(self):
+        """Get all valid figure model targets for this control based on its location in the page hierarchy."""
+        if self.targets:
+            return self.targets
+
+        page = model_manager._get_model_page(self)
+        page_containers = model_manager._get_models(model_type=Container, root_model=page)
+
+        # Find the control's parent model. Set it as the control's parent container it exists.
```

Comment:
I think we might be able to simplify this logic a bit if we move it into `Page`/`Container`, but that also might make a horrible mess. Let me try it out and see.